sw/device/silicon_creator/lib/sigverify/sphincsplus/test/wots_test.c

```diff
@@ -0,0 +1,86 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/sigverify/sphincsplus/wots.h"
+
+#include <stdint.h>
+
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+#include "sw/device/silicon_creator/lib/sigverify/sphincsplus/hash.h"
+#include "sw/device/silicon_creator/lib/sigverify/sphincsplus/params.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+enum {
+  kSpxWotsMsgBytes = ((kSpxWotsLen1 * kSpxWotsLogW + 7) / 8),
+};
+
+static void pk_from_sig_test(const uint8_t *sig, const uint8_t *msg,
+                             const spx_ctx_t *ctx, spx_addr_t *addr,
+                             uint32_t *expected_pk) {
+  uint32_t actual_pk[kSpxWotsPkWords];
+  CHECK(wots_pk_from_sig(sig, msg, ctx, addr, actual_pk) == kErrorOk);
+  if (expected_pk != NULL) {
```

Comment:
If `expected_pk` is expected not to be NULL, you can add an explicit test check for it.